port/osx/omrosdump.c

```diff
@@ -210,25 +211,27 @@ coredump_to_file(mach_port_t task_port, pid_t pid)
 			}
 			if (to_read != data_size) {
 				break;
-			} else {
-				written = pwrite(corefile_fd, (void *)data_read, data_size, seg_file_off + bytes_read);
 			}
+
+			written = pwrite(corefile_fd, (void *)data_read, data_size, seg_file_off + bytes_read);
 			if (written < 0) {
 				perror("pwrite() error writing segment data:");
 				kr = KERN_FAILURE;
 				goto done;
 			}
+			bytes_written += written;
 		}
 
 		segments[i].fileoff = seg_file_off;
+		segments[i].filesize = bytes_written;
```

Comment:
I believe `bytes_written == bytes_read` is true here and so we don't need to introduce `bytes_written` and can instead simply use `bytes_read` here and below.

Reply:
I stopped introducing `bytes_written` as suggested.